.github/workflows/template-build.yml

```diff
@@ -0,0 +1,58 @@
+on:
+  workflow_call:
+    inputs:
+      release-tag-name:
+        required: true
+        type: string
+      push:
+        required: true
+        type: boolean
+env:
+  imageName: "rancher/harvester-pcidevices"
+
+jobs:
+  dapper-build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      id-token: write
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Run dapper
+        run: make ci
+
+      - name: Read some Secrets
+        uses: rancher-eio/read-vault-secrets@main
+        if: ${{ inputs.push == true }}
+        with:
+          secrets: |
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials username | DOCKER_USERNAME ;
+            secret/data/github/repo/${{ github.repository }}/dockerhub/rancher/credentials password | DOCKER_PASSWORD
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v3
+        if: ${{ inputs.push == true }}
+        with:
+          username: ${{ env.DOCKER_USERNAME }}
+          password: ${{ env.DOCKER_PASSWORD }}
+
+      - name: Docker Build
+        uses: docker/build-push-action@v5
+        with:
+          provenance: false
+          context: .
+          platforms: linux/amd64
+          load: true
+          file: package/Dockerfile
+          push: ${{ inputs.push }}
+          tags: ${{env.imageName}}:${{ inputs.release-tag-name }}
+          build-args: |
+            VERSION=${{ github.ref_name }}-${{ github.sha }}-head
```

Comment:
BTW, this one isn't used in Dockerfile as well, so we could remove it.

Reply:
👍